pkg/controllers/vdb/promotesandboxtomain_reconciler.go

```diff
@@ -174,6 +175,8 @@ func (s *PromoteSandboxToMainReconciler) updateSandboxInVdbStatus(ctx context.Co
 			vdbChg.Status.Sandboxes = append(vdbChg.Status.Sandboxes[:i], vdbChg.Status.Sandboxes[i+1:]...)
 			break
 		}
+		// update the annotation to show promote_sandbox has been done
+		vdbChg.Annotations[vmeta.ReplicatedUpgradeSandboxPromotedAnnotation] = vmeta.SandboxPromotedTrue
```

Comment:
I think this should be in `replicatedupgrade_reconciler` in the promotesandbox function, that's the only place we need it. And this function is to update the status  so not a good place to update the annotation.